src/App.jsx

```diff
@@ -14,33 +17,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_RESET = '';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function sortGoods(goods, sortField = '') {
+  const visibleGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case SORT_FIELD_LENGTH:
+        visibleGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        throw new Error('Sort method is unknown!');
+    }
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return visibleGoods;
+}
+
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  let visibleGoods = sortGoods(goodsFromServer, sortField);
+
+  if (reverse) {
+    visibleGoods = visibleGoods.reverse();
+  }
```

Comment:
```suggestion
  if (reverse) {
    visibleGoods.reverse();
  }
```
- You can move reverse logic to the function as well
- Prefer const over let